www/src/components/code-block/index.js

```diff
@@ -0,0 +1,96 @@
+import React from "react"
+import Highlight, { defaultProps } from "prism-react-renderer"
+
+import Copy from "../copy"
+import normalize from "./normalize"
+import { fontSizes, radii, space } from "../../utils/presets"
+
+const getParams = (name = ``) => {
+  const [lang, params = ``] = name.split(`:`)
+  return [
+    lang
+      .split(`language-`)
+      .pop()
+      .split(`{`)
+      .shift(),
+  ].concat(
+    params.split(`&`).reduce((merged, param) => {
+      const [key, value] = param.split(`=`)
+      merged[key] = value
+      return merged
+    }, {})
+  )
+}
+
+/*
+ * MDX passes the code block as JSX
+ * we un-wind it a bit to get the string content
+ * but keep it extensible so it can be used with just children (string) and className
+ */
+export default ({
+  children,
+  className = children.props ? children.props.className : ``,
+}) => {
+  const [language, { title = `` }] = getParams(className)
```

Comment:
@DSchau I love it 😍 and plan to use some of that to add that nice copy feature to coding4.gaiama.org too 👍 🚀 

By the way `mdx` passes additional space separated props as well so you could get the `title` from `props.title` if you pass the title like so
```md
```js title=gatsby-config.js
```
instead of
```md
```js:title=gatsby-config.js
```
then it won't end up in className and you wouldn't have to extract

Yet it's working as is and would be quite a pain to migrate all docs I guess 😅

Reply:
FYI: i found your described behaviour here in the mdx docs:

https://mdxjs.com/guides/live-code#code-block-meta-string

Reply:
Uh yeah thanks for the link that's where I know this from 😁🙏